Aardvark/ARKEmailBugReporter.m

```diff
@@ -37,6 +37,82 @@ @interface ARKInvisibleView : UIView
 @end
 
 
+@interface CALayer (HierarchyDescription)
+
+- (NSString *)_recursiveLayerHierarchyDescriptionWithIndentationLevel:(int)indentationLevel;
+
+@end
+
+
+@implementation CALayer (HierarchyDescription)
+
+- (NSString *)_recursiveLayerHierarchyDescriptionWithIndentationLevel:(int)indentationLevel;
+{
+    NSMutableString *const mutableDescription = [NSMutableString new];
+    for (int i = 0 ; i < indentationLevel ; i++) {
+        [mutableDescription appendString:@"   | "];
+    }
+    
+    [mutableDescription appendFormat:@"%@\n", [self description]];
+    
+    for (CALayer *sublayer in self.sublayers) {
+        [mutableDescription appendString:[sublayer _recursiveLayerHierarchyDescriptionWithIndentationLevel:(indentationLevel + 1)]];
```

Comment:
Totally doesn't matter but it would be a bit more efficient to pass the mutable string into the method as a parameter.